crypto/fipsmodule/bn/montgomery.c

```diff
@@ -418,6 +447,82 @@ static int bn_mod_mul_montgomery_fallback(BIGNUM *r, const BIGNUM *a,
   return ret;
 }
 
+
+#if defined(OPENSSL_BN_ASM_MONT)
+
+// Perform montgomery multiplication using s2n-bignum functions. The arguments
+// are equivalent to the arguments of bn_mul_mont.
+// montgomery_s2n_bignum_mul_mont works only if num is a multiple of 8. For
+// num = 32 or num = 16, this uses faster primitives in s2n-bignum.
+// Additionally, montgomery_s2n_bignum_mul_mont allocates arrays at a stack, and
+// large num leads to out of bounds accesses of the arrays.
+// montgomery_use_s2n_bignum(num) must be called in advance to check these
+// conditions.
+static void montgomery_s2n_bignum_mul_mont(BN_ULONG *rp, const BN_ULONG *ap,
+                                           const BN_ULONG *bp,
+                                           const BN_ULONG *np,
+                                           const BN_ULONG *n0, size_t num) {
+
+#if defined(BN_MONTGOMERY_USE_S2N_BIGNUM)
+
+  // t is a temporary buffer used by big-int multiplication.
+  // bignum_kmul_32_64 requires 96 words.
+  uint64_t t[96];
```

Comment:
if `96` is a constant related to the internals of s2n-bignum than I think it should be defined as a macro in an s2n-bignum header file. Otherwise, s2n-bignum internals could change and we'll forget to change the appropriate values in aws-lc.

Reply:
Addressed it in the latest commit. For each Karatsuba multiplication function (16x16 and 32x32 square/mults), a macro designating the temporary buffer word size is defined and used. To avoid usage of `max()` (which actually doesn't seem to exist in `stdlib.h` or `math.h`..? C is a weird language) an assertion that checks `S2NBIGNUM_KMUL_32_64_TEMP_NWORDS` is the largest one is added to this checker function